Assets/Src/AuthController/CredentialProviders/Firebase/Google/GoogleCredentialProvider.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Threading.Tasks;
+using Src.AuthController.AuthKeys;
+using Src.AuthController.CredentialProviders.Firebase.Google.GoogleRestRequestsAdapter;
+using Src.AuthController.CredentialProviders.Firebase.Google.TokenValidator;
+using Src.AuthController.CredentialProviders.Firebase.Google.UrlOpening;
+using Src.AuthController.REST.PortListener;
+using Src.AuthController.REST.REST_Request_Proxies;
+using Src.AuthController.REST.REST_Response_DTOs;
+
+namespace Src.AuthController.CredentialProviders.Firebase.Google
+{
+    public class GoogleCredentialProvider : IGoogleCredentialProvider
+    {
+        private readonly IGoogleAccessTokenValidator _accessTokenValidator;
+        private readonly IGoogleRestRequestsAdapter _restRequestsAdapter;
+        private readonly IGoogleOAuthUrl _oAuthUrl;
+        private readonly ILocalHttpPortListener _localHttpPortListener;
+
+        private string _authCode;
+        
+        private GoogleIdTokenResponse _googleApiResponse;
+        private float _googleApiResponseIssueTime;
+
+        public GoogleCredentialProvider(IGoogleAccessTokenValidator accessTokenValidator, IGoogleRestRequestsAdapter restRequestsAdapter, IGoogleOAuthUrl oAuthUrl, ILocalHttpPortListener localHttpPortListener)
```

Comment:
It may be more handy to create some general interface for opening urls and use it's implementation in this class. Like IUrlOpener with method Open(string url). By doing so we will not only isolate our GoogleCredentialProvider from url opening logic but also obtain reusable piece of code that may be useful in the future.